Observer/SalesOrderCreditmemoSaveAfter.php

```diff
@@ -61,8 +61,19 @@ public function execute(Observer $observer)
     {
         $creditMemo = $observer->getEvent()->getCreditmemo();
         $creditMemoId = $creditMemo->getId();
+        $balancepayRefundData = $this->balancepayRefundFactory->create()->getCollection()
+            ->addFieldToFilter('credit_memo_id', ['eq' => $creditMemoId])->getData();
+        if (count($balancepayRefundData)>0) {
+            return;
+        }
         $total = $creditMemo->getBaseGrandTotal();
         $invoiceId = $creditMemo->getInvoiceId();
+        $comments = $creditMemo->getComments();
+        $reason = 'Other';
+        foreach ($comments as $comment) {
+            $reason = $comment->getComment();
```

Comment:
what is comments?

Reply:
Comments is the reason value from Dropdown

Reply:
why not reason?

Reply:
We have replace the textbox for comments with Reason Dropdown so had to use the comments input here.